src/Bicep.Core/Semantics/SymbolHierarchy.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.TypeSystem;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bicep.Core.Semantics
+{
+    public class SymbolHierarchy
```

Comment:
What does it mean for a symbol to be a parent/child of another?

Reply:
It's a hierarchy. `FileSymbol` is a parent of all the global declarations, for example.

---
In reply to: [582221724](https://github.com/Azure/bicep/pull/1521#discussion_r582221724) [](ancestors = 582221724)

Reply:
After our discussion I started to think more on this. I'm starting to view it as less a hierarchy of symbols but more of a hierarchy of semantically relevant things with some of them being symbols. I'm also starting to realize that too many things are called Symbols in the semantic model.

---
In reply to: [582234192](https://github.com/Azure/bicep/pull/1521#discussion_r582234192) [](ancestors = 582234192,582221724)